webapp/app.py

```diff
@@ -77,18 +80,17 @@ def load_user_from_request(request: Request) -> User | None:
         # get user data if new session or refresh user data when access token expires
         if oidc_access_token != session.get("oidc_access_token"):
             oidc_data = parse_oidc_data(oidc_jwt_data, options={"verify_exp": False})
-            user = User(
-                id=oidc_data["mit_id"],
-                name=oidc_data["name"],
-                email=oidc_data["preferred_username"],
-            )
             session.update(
-                {
-                    "oidc_access_token": oidc_access_token,
-                    "user": user.__dict__,
-                }
+                {"oidc_access_token": oidc_access_token, "oidc_data": oidc_data}
```

Comment:
I like this a lot.  

I think storing the token as-is, makes perfect sense; easy to compare.  And then we can store the fully parsed OIDC data.